shared/chat/conversation/messages/system-change-retention/index.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import moment from 'moment'
+import * as React from 'react'
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import UserNotice from '../user-notice'
+import {formatTimeForMessages} from '../../../../util/timestamp'
+import * as RPCChatTypes from '../../../../constants/types/rpc-chat-gen'
+
+type Props = {|
+  canManage: boolean,
+  isTeam: boolean,
+  isInherit: boolean,
+  membersType: RPCChatTypes.ConversationMembersType,
+  onClickUserAvatar: (username: string) => void,
+  onManageRetention: () => void,
+  policy: RPCChatTypes.RetentionPolicy,
+  user: string,
+  you: string,
+  timestamp: number,
+|}
+
+const ManageComponent = (props: {canManage: boolean, onManageRetention: () => void}) => {
+  const textType = 'BodySmallSemiboldPrimaryLink'
+  if (props.canManage) {
+    return (
+      <Kb.Text onClick={props.onManageRetention} type={textType}>
+        Manage this
+      </Kb.Text>
+    )
+  }
+  return ''
+}
+
+const connectedUsernamesProps = {
+  colorFollowing: true,
+  inline: true,
+  onUsernameClicked: 'profile',
+  type: 'BodySmallSemibold',
+  underline: true,
+}
+
+const YouOrUsername = (props: {username: string, you: string}) => {
+  if (props.username === props.you) {
+    return 'You'
+  }
+  return <Kb.ConnectedUsernames {...connectedUsernamesProps} usernames={[props.username]} />
+}
+
+const AppliesTo = (props: {membersType: RPCChatTypes.ConversationMembersType, isTeam: boolean}) => {
+  switch (props.membersType) {
+    case RPCChatTypes.commonConversationMembersType.team:
+      if (props.isTeam) {
+        return 'team'
+      } else {
+        return 'channel'
+      }
+  }
+  return 'conversation'
+}
+
+const InheritDescription = (props: {isInherit: boolean}) => {
```

Comment:
i don't think all these components are really necessary, just put this into the parent or  you can just figure this out in the render and render it inline like
```
const inherit = props.isInherit ? ' ' : ''
...
{inherit}
...
```